napari/_qt/qt_qslider_popup.py

```diff
@@ -0,0 +1,129 @@
+        if width > self.min_width:
+            # don't ever make the label smaller ... it causes visual jitter
+            self.min_width = width
+        self.setFixedWidth(width)
+
+    def update_position(self):
+        x = self.get_pos() * 4 - self.width() / 2
+        y = 8 + 6
+        self.move(QPoint(x, -y) + self.slider.pos())
+
+    def mouseDoubleClickEvent(self, event):
+        self.selectAll()
+
+
+class QSliderPopup(QtPopup):
+    def __init__(self, parent=None, horizontal=True, precision=0, **kwargs):
+        """A popup window that contains a range slider and linked LineEdits.
+
+        Parameters
+        ----------
+        parent : QWidget, optional
+            Will like be an instance of QtLayerControls.  Note, providing
+            parent can be useful to inherit stylesheets.
+        horizontal : bool, optional
+            Whether the slider is oriented horizontally, by default True.
+            (Vertical orientation has not been tested much)
+        precision : int, optional
+            Number of decimal values in the labels, by default 0
+        **kwargs
+            all additional keyword arguments will be passed to the RangeSlider
+        """
+        super().__init__(parent)
+        self.precision = precision
+
+        # create slider
+        self.slider = (
+            QSlider(Qt.Horizontal) if horizontal else QSlider(Qt.Vertical)
+        )
+        self.slider.setMinimum(2)
+        self.slider.setMaximum(200)
+        self.slider.setSingleStep(1)
+        self.slider.setValue(100)
+        self.slider.setMinimumHeight(18)
+        self.slider.setFocus()
+        self.slider.valueChanged.connect(self._on_value_change)
+        # self.slider.resized.connect(self._update_cur_label_positions) # no resized signal yet
+
+        # create "floating" value label
+        cval = self.slider.value()
+        get_val_pos = self.slider.value  # TODO: validate
+        self.cval_label = LabelEdit(self._numformat(cval), self, get_val_pos)
+        self.cval_label.editingFinished.connect(self._curval_label_changed)
+        self.cval_label.setToolTip("current gamma")
+
+        # add widgets to layout
+        self.layout = QHBoxLayout()
+        self.frame.setLayout(self.layout)
+        self.frame.setContentsMargins(0, 8, 0, 0)
+        self.layout.addWidget(self.slider, 50)
+
+    def _numformat(self, number):
+        if round(number) == number:
+            return "{:.{}f}".format(number, 0)
+        else:
+            return "{:.{}f}".format(number, self.precision)
+
+    def _update_cur_label_positions(self):
+        self.cval_label.update_position()
+
+    def _on_value_change(self, value):
+        with qt_signals_blocked(self.slider):
+            self.cval_label.setText(str(float(value) / 100.0))
+            self._update_cur_label_positions()
+
+    def _curval_label_changed(self):
+        cval = int(self.curmin_label.text())  # TODO: check int casting here
```

Comment:
for maximum reusability, it might be worth considering implementing self.slider as a "double" slider behind the scenes, so that it can be used stuff like gamma as well.  (Then you would stick with casting to `float` in `_label_changed`, and use `_numformat` to make integer sliders "look" like integer sliders).

<details>
<summary>abbreviated QDoubleSlider Example</summary>

```python
class QDoubleSlider(QSlider):
    """A Slider Widget that can handle float values."""

    PRECISION = 1000

    def __init__(self, parent=None):
        super().__init__(Qt.Horizontal, parent=parent)
        self.setMaximum(10)

    def value(self):
        """Get the slider value and convert to float."""
        return super().value() / self.PRECISION

    def setValue(self, value):
        """Set integer slider position from float ``value``."""
        super().setValue(value * self.PRECISION)

    def setMaximum(self, value):
        """Set maximum position of slider in float units."""
        super().setMaximum(value * self.PRECISION)
```
</details>

Reply:
i agree, i was thinking to do same

Reply:
i agree, i was thinking to do same